### PR TITLE
DCES-193 FDC Global Update

### PR DESCRIPTION
**Pull request by LAA-Dave-Edwards**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/DCES-193)

Adding endpoint to be used in the modernised maat-5 job, the "global update" to prime fdc entries for sending to the debt recovery company.

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Review by stone-dj — CHANGES_REQUESTED** [review]
Have left some review feedback, please take a look and let me know

File: maat-court-data-api/src/main/java/gov/uk/courtdata/repository/FdcContributionsRepository.java [hunk 1]
```diff
@@ -12,4 +14,15 @@
 public interface FdcContributionsRepository extends JpaRepository<FdcContributionsEntity, Integer> {
     List<FdcContributionsEntity> findByStatus(FdcContributionsStatus status);
     List<FdcContributionsEntity> findByIdIn(Set<Integer> ids);
+
+    @Modifying
+    @Query(nativeQuery = true, value = "MERGE into TOGDATA.fdc_contributions fc " +
```
[the remark this answers is not in the document]
Comment (LAA-Dave-Edwards) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/repository/FdcContributionsRepository.java:19]: This is very handy! Thanks for the info on this :)

File: maat-court-data-api/src/test/java/gov/uk/courtdata/dces/service/DebtCollectionRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        when(jdbcTemplate.batchUpdate(mergeSQLCaptor.capture()))
+                .thenReturn(new int[]{1});
+        debtCollectionRepository.globalUpdatePart2();
+
+
+        assertNotNull(mergeSQLCaptor);
+        Assertions.assertArrayEquals(fdcMerge2StatementHash,getCaptorHash());
+    }
+
+    byte[] getCaptorHash(){
```
[the remark this answers is not in the document]
Comment (LAA-Dave-Edwards) [reply maat-court-data-api/src/test/java/gov/uk/courtdata/dces/service/DebtCollectionRepositoryTest.java:55]: I had a think on this, and I'm not sure of the benefit. The int[] isn't too readable in the final static, but neither would a string.
I would argue a {1,-24,-114,53} is just as readable as "a1fc92", not usefully so being my reading. And a change in either would require a computer to notice 100% of the time. 
E.g. aaa213561322231acf -> aaa213561322331acf
Is already painful to notice.

[the remark this answers is not in the document]
Comment (stone-dj) [reply maat-court-data-api/src/test/java/gov/uk/courtdata/dces/service/DebtCollectionRepositoryTest.java:55]: Sure, you make a fair point, no requirement to change it.

**Review by stone-dj — APPROVED** [review]
Changes look good

**Pull request merged**